core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1571,6 +1571,28 @@ class SparkContext(config: SparkConf) extends Logging {
   def killExecutor(executorId: String): Boolean = killExecutors(Seq(executorId))
 
   /**
+   * :: DeveloperApo ::
+   * Request that the cluster manager kill all executors on the specified host.
+   *
+   * Note: This is an indication to the cluster manager that the application wishes to adjust
+   * its resource usage downwards. If the application wishes to replace the executor it kills
```

Comment:
why adjust downwards?  I would have expected kill and replace.  That is what we want for blacklisting, anyway.